### PR TITLE
Fix doctype check issue for metadata fetching. Fixes #4468

### LINKED ISSUE #4468: [Bug]: Missing URL thumbnail
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

The following link has a missing thumbnail:

https://thebetter.news/amazon-deforestation-reduced/

### Steps to Reproduce

1. Create a new post with URL = https://thebetter.news/amazon-deforestation-reduced/
2. Observe missing thumbnail

Expected behavior: Lemmy should extract the jpg from the following document meta tag:

```html
<meta property="og:image" content="https://thebetter.news/wp-content/uploads/2024/01/lula-amazonas.jpg" />
```

resulting in https://thebetter.news/wp-content/uploads/2024/01/lula-amazonas.jpg.

### Technical Details

Observe missing thumbnails:

- https://lemmy.world/post/12218793
- https://lemm.ee/post/24404264

### Version

0.19.3

### Lemmy Instance URL

lemm.ee, lemmy.world

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

File: crates/api_common/src/request.rs [hunk 1]
```diff
@@ -53,7 +53,9 @@ pub async fn fetch_link_metadata(
   // https://github.com/LemmyNet/lemmy/issues/1964
   let html_bytes = response.bytes().await.map_err(LemmyError::from)?.to_vec();
 
-  let opengraph_data = extract_opengraph_data(&html_bytes, url).unwrap_or_default();
+  let opengraph_data = extract_opengraph_data(&html_bytes, url)
+    .map_err(|e| warn!("{e}"))
```
Comment (dessalines) [inline crates/api_common/src/request.rs:57]: Added logging here.

Comment (Nutomic) [reply crates/api_common/src/request.rs:57]: I would put this under info or even verbose. No reason to spam server logs with warnings about this.

File: crates/api_common/src/request.rs [hunk 2]
```diff
@@ -96,7 +98,7 @@ fn extract_opengraph_data(html_bytes: &[u8], url: &Url) -> Result<OpenGraphData,
     .ok_or(LemmyErrorType::NoLinesInHtml)?
     .to_lowercase();
 
-  if !first_line.starts_with("<!doctype html>") {
+  if !first_line.starts_with("<!doctype html") {
     Err(LemmyErrorType::SiteMetadataPageIsNotDoctypeHtml)?
   }
```
Comment (Nutomic) [inline crates/api_common/src/request.rs:103]: Just noticed this change after reading https://github.com/LemmyNet/lemmy/issues/4468. Do we actually need this check at all? Why not attempt `HTML::from_string` always and let it throw an error if its invalid? A check for max file size would make more sense.

File: crates/api_common/src/request.rs (hunk 1 above)
Comment (dessalines) [reply crates/api_common/src/request.rs:57]: K, will use verbose.

**Review by Nutomic — APPROVED** [review]

**Pull request merged**